yellowbrick/text/postag.py

```diff
@@ -114,251 +115,381 @@ def __init__(
-                        self.pos_tag_counts_["infinitive"] += 1
-                    elif tag == "UH":
-                        self.pos_tag_counts_["interjection"] += 1
-                    elif tag == "SYM":
-                        self.pos_tag_counts_["symbol"] += 1
-                    else:
-                        self.pos_tag_counts_["other"] += 1
+        if self.stack:
+            for tagged_doc,y in zip(X,y):
+                for tagged_sent in tagged_doc:
+                    for _, tag in tagged_sent:
+                        if tag.startswith("N"):
+                            self.pos_tag_counts_[y]["noun"] += 1
+                        elif tag.startswith("J"):
+                            self.pos_tag_counts_[y]["adjective"] += 1
+                        elif tag.startswith("V"):
+                            self.pos_tag_counts_[y]["verb"] += 1
+                        # include particles with adverbs
+                        elif tag.startswith("RB") or tag == "RP":
+                            self.pos_tag_counts_[y]["adverb"] += 1
+                        elif tag.startswith("PR"):
+                            self.pos_tag_counts_[y]["pronoun"] += 1
+                        elif tag.startswith("W"):
+                            self.pos_tag_counts_[y]["wh- word"] += 1
+                        elif tag == "CC":
+                            self.pos_tag_counts_[y]["conjunction"] += 1
+                        elif tag == "CD":
+                            self.pos_tag_counts_[y]["digit"] += 1
+                        # combine predeterminer and determiner
+                        elif tag in ["DT" or "PDT"]:
+                            self.pos_tag_counts_[y]["determiner"] += 1
+                        elif tag == "EX":
+                            self.pos_tag_counts_[y]["existential"] += 1
+                        elif tag == "FW":
+                            self.pos_tag_counts_[y]["non-English"] += 1
+                        elif tag == "IN":
+                            self.pos_tag_counts_[y]["preposition"] += 1
+                        elif tag == "POS":
+                            self.pos_tag_counts_[y]["possessive"] += 1
+                        elif tag == "LS":
+                            self.pos_tag_counts_[y]["list"] += 1
+                        elif tag == "MD":
+                            self.pos_tag_counts_[y]["modal"] += 1
+                        elif tag in self.punct_tags:
+                            self.pos_tag_counts_[y]["punctuation"] += 1
+                        elif tag == "TO":
+                            self.pos_tag_counts_[y]["infinitive"] += 1
+                        elif tag == "UH":
+                            self.pos_tag_counts_[y]["interjection"] += 1
+                        elif tag == "SYM":
+                            self.pos_tag_counts_[y]["symbol"] += 1
+                        else:
+                            self.pos_tag_counts_[y]["other"] += 1
+        else:
```

Comment:
Would you mind also applying my comments about control flow logic from the other handle method here as well? As you can see, with a complex function like this one, where the choice really only is a huge switch, ensuring that the control flow happens as low as possible becomes increasingly important! 

As in the first function, you should be able to accomplish what you're looking for with only 2 additions and several modifications.